config/m4/compiler.m4

```diff
@@ -17,8 +17,27 @@ AC_ARG_ENABLE(debug,
         [],
         [enable_debug=no])
 AS_IF([test "x$enable_debug" == xyes],
-        [BASE_CFLAGS="-O0 -D_DEBUG $BASE_CFLAGS"],
-        [BASE_CFLAGS="-O3 $BASE_CFLAGS"])
+        [BASE_CFLAGS="-D_DEBUG $BASE_CFLAGS"],
+        [])
+AS_IF([test "x$enable_debug" == xyes],
+        [AS_IF([test "$enable_fast" == none], [enable_fast="0"], [])],
+        [])
+
+#
+# Optimization level
+#
+AC_ARG_ENABLE(fast,
+        AC_HELP_STRING([--enable-fast], [Set optimization level [0-3]]),
+        [],
+        [enable_fast="none"])
+AS_IF([test -z "$enable_fast"], [],
+      [test "$enable_fast" == "yes"], [BASE_CFLAGS="-O3 $BASE_CFLAGS"],
```

Comment:
so by default UCX will be compiled without optimizations?
what is the reason for this change?

Reply:
No, default behavior will not be changed

Reply:
maybe i'm missing something, but by default enable_fast will be empty, which means `[test -z "$enable_fast"]` is true, so no optimization flags will be added...

Reply:
```
AC_ARG_ENABLE(fast,
        AC_HELP_STRING([--enable-fast], [Set optimization level [0-3]]),
        [],                   >>>>>>>>>>>>>> here is action when enable_fast is set
        [enable_fast="none"]) >>>>>>>>>>>>>> here is action when enable_fast is not set
```
in case if user doesn't set enable_fast then optimization level depends from ```enable_debug``` variable (below, line 36)

Reply:
i see. so what is the purpose of `test -z "$enable_fast`? seems like this variable can never be empty - if it's given, it's set to yes/no, and if it's not given it's set to none?
maybe replace "enable_fast=none" by "enable_fast=yes"?

Reply:
technically it is possible, for example if we decide to disable optimization settings based on another flags (for example for any "exotic" compiler), in this case we have to set to enable_fast empty value to suppress -Ox parameters, in other cases this is unnecessary case.

so, remove it? (at least it will simplify ligic)

Reply:
yes, i think it's better to simplify logic and for now work with enable_fast which can be only "yes"
 (default) or "no" (if ./configure --disable-fast was done")
BTW is "disable-fast" a standard terminology? maybe rename it to "disable-compiler-opt" or something else?

Reply:
done: removed empty switch, argument renamed to disable-compiler-opt